src/services/sync-component-with-file/find-config-file-changes.ts

```diff
@@ -19,62 +20,94 @@ const getRemovedFiles = async (
   compassYmlFilesDiffs: CommitFileDiff[],
   event: PushEvent,
 ): Promise<ComponentUnlinkPayload[]> => {
-  return Promise.all(
+  const settledPromises = await Promise.allSettled(
     compassYmlFilesDiffs.map((diff: CommitFileDiff) => {
-      return getFileContent(token, event.project.id, diff.old_path, event.before).then((componentYaml) => ({
-        componentYaml,
-        filePath: `/${diff.new_path}`,
-        immutableLocalKeyPrefix: event.project.id.toString(),
-      }));
+      return getFileContent(token, event.project.id, diff.old_path, event.before)
+        .then((componentYaml) => ({
+          componentYaml,
+          filePath: `/${diff.new_path}`,
+          immutableLocalKeyPrefix: event.project.id.toString(),
+        }))
+        .catch((err) => {
```

Comment:
With this `catch` block here, would all promises still count as `fulfilled`?

Now that we're using `allSettled`, would we want the error to bubble app and mark the status as `rejected` then?

Reply:
now throwing